### PR TITLE
Add `overridesBaseUrl` to `LanguageOptions`

### PR DESCRIPTION
**Pull request by nf-s**
### What this PR does

Add `overridesBaseUrl` to `LanguageOptions`. This can be used to set the base URL for language override namespace translation files.

For example, if `overridesBaseUrl = "test/path/"`, then the full path for translation override files will be `"test/path/{{lng}}.json"`

#### Geo-rapp config

https://terria-catalogs-public.storage.googleapis.com/geo-rapp/map-config/dev.json has:

```json5
...,
"languageConfiguration": {
  "debug": false,
  "fallbackLanguage": "en",
  "languages": {
    "en": "english"
  },
  "overridesBaseUrl": "https://terria-catalogs-public.storage.googleapis.com/geo-rapp/languageOverrides/dev/"
},
...
```

This means the following english language translation override file will be loaded:

https://terria-catalogs-public.storage.googleapis.com/geo-rapp/languageOverrides/dev/en.json

The file contains the following

```json5
{
  "helpPanel": {
    "menuPaneBody": "Find useful tips on how to use RaPP Map by either checking the guides below or by contacting the team at [geoglam.rapp at csiro.au](mailto:geoglam.rapp@csiro.au)"
  },
...
```

Which you will see in the help panel for the following test link

http://ci.terria.io/overrides-base-url/#configUrl=https://terria-catalogs-public.storage.googleapis.com/geo-rapp/map-config/dev.json

### Checklist

- [ ] ~There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)~
- [x] I've updated relevant documentation in `doc/`.
- [x] I've updated CHANGES.md with what I changed.
- [x] I've provided instructions in the PR description on how to test this PR.

### CODE REVIEW

File: doc/customizing/client-side-config.md [hunk 1]
```diff
@@ -162,14 +162,15 @@ Configuration of items to appear in the search bar
...
+| fallbackLanguage          | yes      | **string**       | `"en"`                                                    | Fallback language used if contents are not available in the currently selected language.                                                                                                                                                                                        |
+| changeLanguageOnStartWhen | yes      | **string[]**     | `["querystring", "localStorage", "navigator", "htmlTag"]` | Order of user language detection. See [i18next browser language detection documentation](https://github.com/i18next/i18next-browser-languageDetector) for details.                                                                                                              |
+| overridesBaseUrl          | yes      | **string**       |                                                           | Base URL for override namespace translation files. If set, this make up the base URL for translation override files. Should end in "/". For example, if `overridesBaseUrl = "test/path/"`, then the full path for translation override files will be `"test/path/{{lng}}.json"` |
```
Comment (staffordsmith83) [inline doc/customizing/client-side-config.md]: Should be "This makes up..."?

File: lib/Models/Internationalization.ts [hunk 2]
```diff
@@ -33,6 +34,12 @@ export interface LanguageConfiguration {
   languages: Object;
   fallbackLanguage: string;
   changeLanguageOnStartWhen: string[];
+
+  /** Base URL for override namespace translation files. If set, this make up the base URL for translation override files. Should end in /
```
Comment (staffordsmith83) [inline lib/Models/Internationalization.ts]: Should be "this makes up..."?

**Review by staffordsmith83 — COMMENTED** [review]
Looks good just a couple of spelling erros then will approve! Handy feature

File: lib/Models/Internationalization.ts (hunk 2 above)
Comment (nf-s) [reply lib/Models/Internationalization.ts]: Done!

File: doc/customizing/client-side-config.md (hunk 1 above)
Comment (nf-s) [reply doc/customizing/client-side-config.md]: Done!

**Review by staffordsmith83 — APPROVED** [review]
Looks good!

**Pull request merged**